src/components/Crackmapexec/Crackmapexec.tsx

```diff
@@ -6,38 +6,78 @@ import ConsoleWrapper from "../ConsoleWrapper/ConsoleWrapper";
 import { UserGuide } from "../UserGuide/UserGuide";
 import { SaveOutputToTextFile_v2 } from "../SaveOutputToFile/SaveOutputToTextFile";
 
-const title = "Crackmapexec Tool";
-const description_userguide =
-    "The Crackmapexec tool withholds a package which is a swiss army knife used for penetration testing Windows or " +
-    "Active Directory environments. The tool is capable of enumerating information from logged on users, perform psexec " +
-    "styled attacks through SMB spidering, auto-injecting Shellcode into memory through Powershell, and dumping NTDS.nit. " +
-    "Any used or dumped credentials will be stored to a database.\n\nFurther information can be found at: https://www. " +
-    "kali.org/tools/crackmapexec/\n\n" +
-    "Using Crackmapexec:\n" +
-    "Step 1: Enter a Target IP address.\n" +
-    "       Eg: 192.168.0.1\n\n" +
-    "Step 2: Enter a Username.\n" +
-    "       Eg: admin\n\n" +
-    "Step 3: Enter a Password.\n" +
-    "       Eg: admin\n\n" +
-    "Step 4: Click Start Searching to commence Crackmapexec's operation.\n\n" +
-    "Step 5: View the Output block below to see the results of the tool's execution.";
-
-interface FormValues {
+/**
+ * Interface representing the values in the Crackmapexec form.
+ * 
+ * @property {string} ip - The target IP address.
+ * @property {string} username - The username for authentication.
+ * @property {string} password - The password for authentication.
+ * @property {number} timeout - The time (in seconds) to wait for a response. Default is 60.
+ */
+interface FormValuesType {
     ip: string;
     username: string;
     password: string;
     timeout: number;
 }
 
+// Section: Constants
+
+/** Tool name as per its source documentation. */
+const title = "Crackmapexec";
+
+/** 
+ * Description text used in the user guide of the component.
+ * The Crackmapexec tool helps in penetration testing for Windows/Active Directory environments.
+ * 
+ * Steps:
+ * 1. Enter the target IP address.
+ * 2. Enter the username.
+ * 3. Enter the password.
+ * 4. Click "Start Searching" to execute the Crackmapexec tool.
+ * 5. Review the output results in the Output block.
+ * 
+ * Further information can be found at: https://www.kali.org/tools/crackmapexec/
+ */
+const descriptionUserGuide =
+    "Crackmapexec is a post-exploitation tool that helps in penetration testing for Windows/Active Directory environments. " +
+    "It is capable of enumerating users, shares, and computers, executing commands, and dumping credentials from NTDS.dit. " +
+    "It automates various tasks to aid in testing the security of a network.\n\n" +
+    "Steps:\n" +
+    "1. Enter the target IP address.\n" +
+    "2. Enter the username.\n" +
+    "3. Enter the password.\n" +
+    "4. Click 'Start Searching' to execute the Crackmapexec tool.\n" +
+    "5. Review the output results in the Output block.\n\n" +
+    "Further information can be found at: https://www.kali.org/tools/crackmapexec/";
```

Comment:
The source link should be its own constant that contains the link to the source document.